src/main/java/org/team1540/bunnybotTank2023/io/turret/TurretIOReal.java

```diff
@@ -0,0 +1,53 @@
+package org.team1540.bunnybotTank2023.io.turret;
+
+import com.ctre.phoenix6.configs.Slot0Configs;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.ForwardLimitValue;
+import com.ctre.phoenix6.signals.ReverseLimitValue;
+import edu.wpi.first.math.geometry.Rotation2d;
+import org.team1540.bunnybotTank2023.Constants;
+
+public class TurretIOReal implements TurretIO {
+
+    // fields ^-^
+    private final TalonFX motor = new TalonFX(Constants.TurretConstants.MOTOR_ID);
+    final PositionVoltage positionVoltage = new PositionVoltage(0);
+    private Rotation2d setPosition = new Rotation2d();
+
+    // constructor
+    TurretIOReal() {
+        // robot init, set slot 0 gains
+        var slot0Configs = new Slot0Configs();
+        slot0Configs.kP = Constants.TurretConstants.kP;
+        slot0Configs.kI = Constants.TurretConstants.kI;
+        slot0Configs.kD = Constants.TurretConstants.kD;
+
+        motor.getConfigurator().apply(slot0Configs, 0.050);
```

Comment:
In addition to configuring PID constants, you will also have to configure motor inversion, brake mode, and current limits here. The configuration can get pretty cluttered with so many parameters, so you might want to look into putting these configs into a separate file, such as in [Pesto's code](https://github.com/flamingchickens1540/pesto/blob/main/src/main/java/org/team1540/robot2023/CTREConfigs.java), or, more recently, in the [shooter code](https://github.com/flamingchickens1540/bunnybots-delphi-2023/blob/alvin/shooter/src/main/java/org/team1540/bunnybotTank2023/CTREConfigs.java)